src/il/tos.nim

```diff
@@ -178,35 +181,37 @@ proc `$`*(self: Pattern): string =
     of PatternKind.UnderScore:
         "_"
 proc `$`*(self: IdentDef): string =
-    case self.kind
-    of DefKind.Def:
-        let
-            pat = $self.pat
-            typ = if self.typ.isNone: "" else: fmt": {self.typ.get}"
-            default = if self.default.isNone: "" else: fmt" = {self.default.get}"
-            docStr = if self.docStr.isNone: "" else: fmt" #{self.docStr.get}"
-        pat & typ & default & docStr
-    of DefKind.Comment:
-        "#" & self.comment
+    let
+        pat = $self.pat
+        typ = if self.typ.isNone: "" else: fmt": {self.typ.get}"
+        default = if self.default.isNone: "" else: fmt" = {self.default.get}"
+        docStrs = self.comments.filterIt(it.isDoc)
```

Comment:
In this case,
```
let
  a = 3
  ## aa
  # a
  ## bb
```

Reply:
i think it works correctly (or my opinion that is stated in the above conversation is incorrect).